plugins/c8y_remote_access_plugin/src/proxy.rs

```diff
@@ -0,0 +1,53 @@
+use async_compat::CompatExt;
+use futures::future::join;
+use futures::future::select;
+use futures_util::io::AsyncReadExt;
+use futures_util::io::AsyncWriteExt;
+use tokio::net::TcpStream;
+use tokio::net::ToSocketAddrs;
+
+use url::Url;
+
+use crate::auth::Jwt;
+use crate::Websocket;
+
+/// This proxy creates a TCP connection to a local socket and creates a websocket. Cumulocity cloud will initiate a
+/// connection to the websocket. Any data received from the socket is sent out via the websocket and any data received
+/// from the websocket is sent to the local socket.
+pub struct WebsocketSocketProxy {
+    socket: TcpStream,
+    websocket: Websocket,
+}
+
+impl WebsocketSocketProxy {
+    pub async fn connect<SA: ToSocketAddrs + std::fmt::Debug>(
+        url: &Url,
+        socket: SA,
+        jwt: Jwt,
+    ) -> Result<Self, std::io::Error> {
+        let socket_future = TcpStream::connect(socket);
+        let websocket_future = Websocket::new(url, jwt.authorization_header());
+
+        match join(socket_future, websocket_future).await {
+            (Err(socket_error), _) => panic!("{socket_error}"),
+            (_, Err(websocket_error)) => panic!("{websocket_error}"),
```

Comment:
I would consider to use `miette` here to help the user diagnose connection errors - which are quite likely since the socket address is provided by the user. I even wonder if such errors can be forwarded back to the C8Y user?

Reply:
This should definitely use miette, I just overlooked this bit when adding in proper error handling from the prototype.